src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.MetadataDb.cs

```diff
@@ -87,7 +87,7 @@ private struct MetadataDb : IDisposable
             internal int Length { get; private set; }
             private byte[] _data;
 #if DEBUG
-            private readonly bool _isLocked;
+            private bool _isLocked;
```

Comment:
Making this mutable for the new JsonElement.Parse API helps prevent a new alloc here.